GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -554,41 +521,44 @@ void UpdateRow(int row)
         private void UpdateVisibleRowRange()
         {
             var oldRange = _visibleRowRange;
-            var fromIndex = FirstDisplayedCell?.RowIndex ?? 0;
-            var toIndex = _rowHeight > 0 ? fromIndex + (Height / _rowHeight) : fromIndex;
+            var fromIndex = Math.Max(0, FirstDisplayedScrollingRowIndex);
+            var toIndex = _rowHeight > 0 ? fromIndex + (Height / _rowHeight) + 5/*Add 5 for rounding*/ : fromIndex;
+
+            if (fromIndex >= _graphModel.Count)
+            {
+                fromIndex = _graphModel.Count;
```

Comment:
After reverting VisibleRowRange.Count you have to correct this to be 
```csharp
fromIndex = _graphModel.Count - 1;
```
and in the 534 line as well.

Reply:
Done.